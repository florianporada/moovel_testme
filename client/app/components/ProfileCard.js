import React from 'react';
import { Text, TouchableHighlight, View } from 'react-native';
import PropTypes from 'prop-types';

class ProfileCard extends React.Component {
  constructor(props) {
    super(props);
  }

  _setModalVisible(visible) {
    this.setState({modalVisible: visible});
  }

  render() {
    return (
      <View>
       <View>
         <Text>Hello {this.props.name}!</Text>

         <TouchableHighlight onPress={() => {
           this._setModalVisible(false)
         }}>
           <Text>Hide Modal</Text>
         </TouchableHighlight>
       </View>
      </View>
    );
  }
}

ProfileCard.propTypes = {
  name: PropTypes.string,
};


export default ProfileCard;
