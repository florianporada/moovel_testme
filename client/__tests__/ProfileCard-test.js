import 'react-native';
import React from 'react';
import renderer from 'react-test-renderer';

import ProfileCard from '../app/components/ProfileCard';

it('renders correctly', () => {
  const dummyData = {
    "avatar_url": "https://avatars0.githubusercontent.com/u/39191?v=4",
    "bio": null,
    "blog": "http://paulirish.com",
    "company": "Google Chrome, â™¥z",
    "created_at": "2008-12-09T00:39:23Z",
    "email": null,
    "events_url": "https://api.github.com/users/paulirish/events{/privacy}",
    "followers": 24331,
    "followers_url": "https://api.github.com/users/paulirish/followers",
    "following": 244,
    "following_url": "https://api.github.com/users/paulirish/following{/other_user}",
    "gists_url": "https://api.github.com/users/paulirish/gists{/gist_id}",
    "gravatar_id": "",
    "hireable": null,
    "html_url": "https://github.com/paulirish",
    "id": 39191,
    "location": "Palo Alto",
    "login": "paulirish",
    "name": "Paul Irish",
    "organizations_url": "https://api.github.com/users/paulirish/orgs",
    "public_gists": 109,
    "public_repos": 242,
    "received_events_url": "https://api.github.com/users/paulirish/received_events",
    "repos_url": "https://api.github.com/users/paulirish/repos",
    "site_admin": false,
    "starred_url": "https://api.github.com/users/paulirish/starred{/owner}{/repo}",
    "subscriptions_url": "https://api.github.com/users/paulirish/subscriptions",
    "type": "User",
    "updated_at": "2017-09-11T21:30:35Z",
    "url": "https://api.github.com/users/paulirish",
  }

  const tree = renderer.create(
    <ProfileCard modalContent={dummyData} />
    ).toJSON();
  expect(tree).toMatchSnapshot();
});
