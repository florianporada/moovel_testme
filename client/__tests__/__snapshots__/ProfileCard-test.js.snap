// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  style={
    Array [
      Object {
        "flex": 0,
        "flexDirection": "row",
        "paddingBottom": 15,
      },
    ]
  }
>
  <View>
    <View
      activeOpacity={0.7}
      onLongPress={undefined}
      onPress={undefined}
      style={
        Array [
          Object {
            "backgroundColor": "transparent",
            "height": 75,
            "paddingBottom": 10,
            "paddingRight": 10,
            "paddingTop": 10,
            "width": 75,
          },
          Object {
            "borderRadius": 37.5,
          },
          undefined,
        ]
      }
    >
      <View
        style={
          Array [
            Object {
              "alignItems": "center",
              "alignSelf": "stretch",
              "backgroundColor": "rgba(0,0,0,0.2)",
              "bottom": 0,
              "flex": 1,
              "justifyContent": "center",
              "left": 0,
              "position": "absolute",
              "right": 0,
              "top": 0,
            },
            Object {
              "borderRadius": 37.5,
            },
            undefined,
          ]
        }
      >
        <Image
          source={
            Object {
              "uri": "https://avatars0.githubusercontent.com/u/39191?v=4",
            }
          }
          style={
            Array [
              Object {
                "height": 75,
                "width": 75,
              },
              Object {
                "borderRadius": 37.5,
              },
              undefined,
            ]
          }
        />
      </View>
    </View>
    <View
      style={
        Array [
          undefined,
        ]
      }
    >
      <View
        onPress={undefined}
        style={
          Array [
            Object {
              "alignItems": "center",
              "backgroundColor": "#43484d",
              "borderRadius": 20,
              "justifyContent": "center",
              "padding": 12,
              "paddingBottom": 3,
              "paddingTop": 3,
            },
            Array [
              Object {
                "backgroundColor": "#4d86f7",
                "borderRadius": 10,
                "padding": 5,
                "position": "absolute",
                "right": -5,
                "top": -75,
              },
            ],
          ]
        }
      >
        <Text
          accessible={true}
          allowFontScaling={true}
          disabled={false}
          ellipsizeMode="tail"
          style={
            Array [
              Object {
                "color": "white",
                "fontSize": 14,
              },
              Object {
                "color": "#e1e8ee",
              },
            ]
          }
        >
          24k
        </Text>
      </View>
    </View>
  </View>
  <View
    style={
      Object {
        "flex": 1,
        "paddingLeft": 10,
      }
    }
  >
    <Text
      accessible={true}
      allowFontScaling={true}
      disabled={false}
      ellipsizeMode="tail"
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        disabled={false}
        ellipsizeMode="tail"
        style={
          Object {
            "fontWeight": "bold",
          }
        }
      >
        Username: 
      </Text>
      paulirish
    </Text>
    <Text
      accessible={true}
      allowFontScaling={true}
      disabled={false}
      ellipsizeMode="tail"
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        disabled={false}
        ellipsizeMode="tail"
        style={
          Object {
            "fontWeight": "bold",
          }
        }
      >
        Name: 
      </Text>
      Paul Irish
    </Text>
    <Text
      accessible={true}
      allowFontScaling={true}
      disabled={false}
      ellipsizeMode="tail"
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        disabled={false}
        ellipsizeMode="tail"
        style={
          Object {
            "fontWeight": "bold",
          }
        }
      >
        Joined: 
      </Text>
      2008-12-09
    </Text>
    <Text
      accessible={true}
      allowFontScaling={true}
      disabled={false}
      ellipsizeMode="tail"
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        disabled={false}
        ellipsizeMode="tail"
        style={
          Object {
            "fontWeight": "bold",
          }
        }
      >
        Bio: 
      </Text>
    </Text>
  </View>
</View>
`;
